### PR TITLE
[Issue 8346][Admin API] Validate retention policy

### LINKED ISSUE #8346: [Admin API] Prevent setting an invalid Retention Policy
**Problem**
Currently if one [sets in the retention policy](https://pulsar.apache.org/docs/en/2.6.1/cookbooks-retention-expiry/#set-retention-policy) `--time` to some value and sets `--size` to 0, this will lead to the situation where there is no retention. Such retention policy should be rejected since the time setting will have no meaning when size is 0. (the same applies the other way around, setting time to 0 and size to some value)

**Describe the solution you'd like**
An retention policy where either time or size is non-zero and other is zero should be rejected in validation. This would be useful to prevent users creating invalid retention policies. 

**Additional context**
#8345

### PR DESCRIPTION
**Pull request by lhotari**
Fixes #8346

### Motivation

See #8346, #8345

### Modifications

Prevent setting invalid retention policy where either size or time limit is set to zero while the other limit has a non-zero value.
The reason for this is that setting either size or time limit to zero will effectively disable the retention policy.
It is confusing for the end-user if it's possible to set a value for the other limit while it's ignored when the other limit has
the value of zero. This might lead to the incorrect assumption that the value of 0 has the meaning that the limit isn't effective.
The validation will provide the Admin API end user a useful error message if the validation fails.

### CODE REVIEW

**Comment by lhotari** [comment]
since master branch is broken atm, I cherry picked the commit from #8361 to fix the broken state of CI. That PR should be merged before this one.

File: pom.xml [hunk 1]
```diff
@@ -126,7 +126,7 @@ flexible messaging model and an intuitive client API.</description>
     <protobuf3.version>3.11.4</protobuf3.version>
     <protoc3.version>3.11.4</protoc3.version>
     <grpc.version>1.31.0</grpc.version>
-    <protoc-gen-grpc-java.version>1.12.0</protoc-gen-grpc-java.version>
+    <protoc-gen-grpc-java.version>${grpc.version}</protoc-gen-grpc-java.version>
```
Comment (sijie) [inline pom.xml:129]: Is this change related? If not, I'd suggest removing this change from this pull request. Especially we can the way how we define the dependency to using `grpc-bom`. It should be done in a separate PR.

**Comment by lhotari** [comment]
I'm running an experiment to see if reverting 647d3c2 fixes the grpc / protobuf compatibility issues.

File: pom.xml (hunk 1 above)
Comment (lhotari) [reply pom.xml]: yes, this is completely unrelated. I was running the experiment in this PR to see if I could fix the grpc upgrade. I'll remove the changes once #8363 has been merged.

**Review by codelipenghui — APPROVED** [review]
Looks good to me.
